### DISCUSSION TITLE
Is there a way to pass named arguments?

### DISCUSSION THREAD

**Discussion by scullionw** [casey/just#1262] [Q&A] [answered]
For recipe with lots of parameters, it is sometimes unclear what argument you are passing, eg:

`just replay 6200 12013 47257 2022-07-06 18`

It would be nice to be able to `just replay a=6200 b=12013 c=47257 ...`, or `just replay -a 6200 -b 12013 -c 47257 ...`

Does this exist and I just didn't find it in the docs?

**Comment by casey** [comment, marked as answer]
Not yet, unfortunately! See #476 for some ideas, but nothing has been implemented yet.

**Reply by scullionw** [reply]
as an alternative i am now using:
```j
replay a=env_var("a") b=env_var("b") b=env_var("b") date=env_var("date") hour=env_var("hour"):
   echo called with {{a}}
```

which lets the caller do `a=123 b=456 c=789 date=2022-07-07 hour=10 just replay`

it's not great but it still helps readability
